### ISSUE TITLE
Bug: `--ignore-pattern` doesn't work if path starts with `./`

### ISSUE THREAD

**Issue by OliverJAsh** [eslint/eslint#16001]
### Environment

Node version: 16.3.0
npm version: 7.15.1
Local ESLint version: 8.17.0
Global ESLint version: none
Operating System: macOS 12.4

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parserOptions: {
        ecmaVersion: 2017,
    },
    rules: {
        semi: 2,
    },
};

```
</details>

`main.js`:
```js
const a = 1
```

`type-tests/foo.js`:
```js
const a = 1
```

Run `eslint . --ignore-pattern ./type-tests/`.

### What did you expect to happen?

The file `type-tests/foo.js` should be ignored.

### What actually happened?

The file `type-tests/foo.js` is not ignored.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

If I adjust the `--ignore-pattern` path to remove the `./` prefix, it works as expected: `eslint . --ignore-pattern type-tests/`. However, this shouldn't be necessary.

**Comment by OliverJAsh** [comment]
You are right 😄 TIL!
